From 14a720a94e913c149ad8c77a2a5fd53fc6aca8ac Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Fri, 11 Dec 2015 16:18:00 +0530
Subject: [PATCH 1/8] USBC/FUSB: Fix vconn supplied to valid CC instead of
 unused CC.

Vconn should be supplied on unused CC line. But currently it is
getting supplied on valid cc which is used for pd communication.
This patch fixes this issue.

Change-Id: I53154a31b5f6fd21b8437e35d7689ede3b5b3606
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-12118
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Reviewed-on: https://android.intel.com:443/447408
---
 drivers/usb/typec/phy-fusb300.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/typec/phy-fusb300.c b/drivers/usb/typec/phy-fusb300.c
index 36a447d..6fc9be6 100644
--- a/drivers/usb/typec/phy-fusb300.c
+++ b/drivers/usb/typec/phy-fusb300.c
@@ -1121,9 +1121,9 @@ static int fusb300_enable_vconn(struct typec_phy *phy, bool en)
 	mutex_lock(&chip->lock);
 	if (en) {
 		if (phy->valid_cc == TYPEC_PIN_CC1)
-			val = FUSB300_SWITCH0_VCONN_CC1_EN;
-		else if (phy->valid_cc == TYPEC_PIN_CC2)
 			val = FUSB300_SWITCH0_VCONN_CC2_EN;
+		else if (phy->valid_cc == TYPEC_PIN_CC2)
+			val = FUSB300_SWITCH0_VCONN_CC1_EN;
 	}
 
 	ret = regmap_update_bits(chip->map, FUSB300_SWITCH0_REG,
-- 
1.9.1

